ext/page_log/palite/palite.cpp

```diff
@@ -1357,21 +1355,23 @@ class Storage {
     }
 
     uint64_t
-    get_global(Connection &conn, GlobalKey key)
+    get_lsn(Connection &conn, Statement stmt_type)
     {
-        StatementPtr stmt = conn.db_statement(Statement::GET_GLOBAL);
-        SQ_CHECK(sqlite3_bind_int64, stmt.get(), 1, static_cast<sqlite3_int64>(key));
+        StatementPtr stmt = conn.db_statement(stmt_type);
         SQ_CHECK(sqlite3_step, stmt.get());
         return sqlite3_column_int64(stmt.get(), 0);
     }
 
-    void
-    put_global(Connection &conn, GlobalKey key, uint64_t val)
+    uint64_t
+    get_next_lsn(Connection &conn)
     {
-        StatementPtr stmt = conn.db_statement(Statement::PUT_GLOBAL);
-        SQ_CHECK(sqlite3_bind_int64, stmt.get(), 1, static_cast<sqlite3_int64>(key));
-        SQ_CHECK(sqlite3_bind_int64, stmt.get(), 2, static_cast<sqlite3_int64>(val));
-        SQ_CHECK(sqlite3_step, stmt.get());
+        return get_lsn(conn, Statement::GET_NEXT_LSN);
+    }
+
+    uint64_t
+    get_last_lsn(Connection &conn)
```

Comment:
Could it be const (and take a const connection reference)?


```suggestion
    get_last_lsn(const Connection &conn) const
```

Reply:
Same comment as above. Right now the code is not quite clean in terms of const-correctness. Making it in one place will require many changes everywhere.